actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -2535,13 +2535,9 @@ private void assignRoles(Request actorMessage) {
       sender().tell(exception, self());
       return;
     }
-    if (!StringUtils.isBlank(userId)) {
-      esUsrRes =
-          ElasticSearchUtil.getDataByIdentifier(
-              ProjectUtil.EsIndex.sunbird.getIndexName(),
-              ProjectUtil.EsType.user.getTypeName(),
-              userId);
-    } else {
+    //Fetch userId if it is not provided
```

Comment:
1) Remove check on userName.
2) updatedBy and updatedDate needs to be modified.